helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/AssignmentMetadataStore.java

```diff
@@ -92,11 +92,14 @@ protected AssignmentMetadataStore(BucketDataAccessor bucketDataAccessor, String
     return _bestPossibleAssignment;
   }
 
-  public void persistBaseline(Map<String, ResourceAssignment> globalBaseline) {
+  /**
+   * @return true if a new baseline was persisted.
```

Comment:
The java doc is incomplete. 
1. No mention of the input parameter
2. parameter name is inconsistent with method name
3. I feel a more generic `persistAssignment(Map<String, ResourceAssignment> assignments)` is sufficient.  Keep the difference implemented elsewhere

Reply:
The parameter is very obvious, I think saying that I'm passing a global baseline assignment by a parameter named globalBaseline is too verbose.
It is debatable if we should name it baseline or global baseline. I'm fine with both. But it is not the key part of this PR, so I don't want to extend my change's scope. Please feel free to change it.

One generic method is not enough. Unless we add an enum to point which type of assignment it is. I'd prefer to keep this unchanged.